torch/distributed/optim/post_localSGD_optimizer.py

```diff
@@ -65,10 +66,31 @@ def __repr__(self):
         return self.optim.__repr__()
 
     def state_dict(self):
-        return self.optim.state_dict()
+        r"""
+        This is same as torch.optim.Optimizer state_dict,
```

Comment:
Mainly as an FYI: PyTorch follows [Google-style docstrings](https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html) and uses Sphinx to build the docs. This means that we can [cross-reference objects](https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#cross-referencing-python-objects) so that a reference here will link to that object's docs. (For example, if you go [here](https://pytorch.org/docs/stable/distributed.optim.html?highlight=zeroredundancyoptimizer#torch.distributed.optim.ZeroRedundancyOptimizer), you can click on `optim.Optimizer` to go to the `torch.optim.Optimizer` docs or click on `torch.nn.parallel.DistributedDataParallel` to go to the DDP docs.)

It can be a bit gimmicky, so I try to just check the docs created with the PR.

Some general notes:
- Double backticks (e.g. \`\`text\`\`) will render as code type (like `text`) but single backticks will not when you are not doing cross-references
- Cross-references only use single backticks with syntax like :meth:\`method_name\` or :class:\`class_name\`
- Docs will only render for public APIs and if there is a corresponding `.rst` file
- Try to use the cross-references like \`:class:torch.optim.Optimizer\` when appropriate since even if Sphinx cannot link the reference, it will render as code type

Reply:
Thanks for reference! Updated docstrings